File: 백준/Silver/7568. 덩치/덩치.py
import sys 
input = sys.stdin.readline 

n = int(input())
lst = []
for _ in range(n):
    a, b = map(int, input().split())
    lst.append([a, b])

for i in range(n):
    cnt = 1
    for j in range(n):
        if lst[i][0] < lst[j][0] and lst[i][1] < lst[j][1]:
            cnt += 1
    print(cnt, end=" ")

File: 백준/Silver/7568. 덩치/덩치.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

class Main {
    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        int[][] peo = new int[n][];
        for (int i = 0; i < n; i++) {
            String[] input = br.readLine().split(" ");
            peo[i] = new int[]{Integer.parseInt(input[0]), Integer.parseInt(input[1]), i};
        }

        for (int i = 0; i < n; i++) {
            int cnt = 1;
            for (int j = 0; j < n; j++) {
                if (peo[i][0] < peo[j][0] && peo[i][1] < peo[j][1]) cnt++;
            }
            System.out.print(cnt + " ");
        }
    }
}
